### ISSUE TITLE
Manually created SecurityScheme's not retained when openapi.json is created

### ISSUE THREAD

**Issue by stiancognite** [swagger-api/swagger-core#2515]
I have a Jersey application and I configure OpenApi like this:

```
    // SecurityScheme api-key
    SecurityScheme securitySchemeApiKey = new SecurityScheme();
    securitySchemeApiKey.setName("api-key");
    securitySchemeApiKey.setType(SecurityScheme.Type.APIKEY);
    securitySchemeApiKey.setIn(SecurityScheme.In.HEADER);

    OpenApiResource openApiResource = new OpenApiResource();
    OpenAPI openAPI = new OpenAPI();
    Info info = new Info()
        .title("My API")
        .version("0.1")
        .description("This is the documentation for my API");
    openAPI.info(info);
    openAPI.schemaRequirement(securitySchemeApiKey.getName(), securitySchemeApiKey);
    SwaggerConfiguration oasConfig = new SwaggerConfiguration()
        .openAPI(openAPI);
    openApiResource.setOpenApiConfiguration(oasConfig);
    resourceConfig.register(openApiResource);
```

`resourceConfig` is an instance of `org.glassfish.jersey.server.ResourceConfig`. My problem is that the security scheme doesn't end up in the generated openapi.json file. Am I doing something wrong? The information I put into my `Info` class is shown in the resulting openapi.json file.

It looks to me like the call to `ctx.read()` (`GenericOpenApiContext.read()`) in `BaseOpenApiResource.getOpenApi()` returns an `OpenApi` instance that has my `Info` information, but not my security scheme.

I'm using 2.0.0-rc2.

**Comment by frantuma** [comment]
Thanks for reporting this! it's a bug, and is fixed in #2519 which is currently in `io.swagger.core.v3:swagger-jaxrs2:2.0.0-SNAPSHOT` (please notice the `groupId` change) available on sonatype, and will be included in next rc. Closing as fixed, please reopen if you still experience issues.

**Comment by stiancognite** [comment]
That works! Thank you :)
